app/controllers/api/exercises/scratchpad_page_controller.rb

```diff
@@ -0,0 +1,34 @@
+module API
+  module Exercises
+    class ScratchpadPageController < BaseController
+      before_action :use_exercise
+
+      def show
+        page = ScratchpadPage.find_by(about: @exercise, author: current_user)
+
+        render json: SerializeScratchpadPage.(page)
+      end
+
+      def create
+        page = ScratchpadPage.create!(scratchpad_page_params)
+
+        render json: SerializeScratchpadPage.(page)
+      end
+
+      private
+      def use_exercise
+        track = Track.find_by(slug: params[:track_id])
+
+        return render_404(:track_not_found, fallback_url: tracks_url) if track.blank?
+
+        @exercise = track.exercises.find_by(slug: params[:exercise_id])
+
+        return render_404(:exercise_not_found, fallback_url: track_url(track)) if @exercise.blank?
+      end
+
+      def scratchpad_page_params
+        params.require(:scratchpad_page).permit(:content_markdown).merge(about: @exercise, author: current_user)
+      end
+    end
+  end
+end
```

Comment:
So, I think the scratchpad is a whole concept in itself, so shouldn't live inside the exercise namespace.

Instead the URLs should be:
```
GET /api/v1/scratchpad/:category/:page
PATCH /api/v1/scratchpad/:category/:page
```

Rather than use a `resource` block in the routes, I'd probably just have those two lines specified out.

So for this bit you're working on, it would be:
GET `/api/scratchpad/mentoring:exercise/ruby:lasagna`

So `params[:category]` == "mentoring:exercise" and `params[:exercise]` == "ruby:lasagna". Then the category field in the db should be params[:category] and the `about` exercise record can then be found via splitting on the `:`.

For now, you can just do that in the controller (and hard-code it to look up an exercise), and then I'll work out the exact model I want for generalising before we launch, once I've considered all the other types of scratchpad page :)

Reply:
Cool. I've done this.